Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -231,6 +241,33 @@ - (void)createSimulatorWithContext:(BPExecutionContext *)context {
     [context.runner createSimulatorWithDeviceName:deviceName completion:handler.defaultHandlerBlock];
 }
 
+- (void)reuseSimulatorWithContext:(BPExecutionContext *)context {
+    NSString *stepName = REUSE_SIMULATOR(context.attemptNumber);
+    
+    [[BPStats sharedStats] startTimer:stepName];
+    [BPUtils printInfo:INFO withString:stepName];
+    
+    if ([context.runner useSimulatorWithDeviceID: [[NSUUID alloc] initWithUUIDString:context.config.useDeviceID]]) {
+        context.simulatorCreated = YES; //if we don't set this flag, deleteSimulatorWithContext() won't proceed
+        
+        NEXT([self installApplicationWithContext:context]);
```

Comment:
Do you mean just delete he app or reset the entire device?  As far as I know, reset the device will require a reboot, which defeat the purpose of this feature.  To delete the app only should be OK.